app/javascript/components/maps/components/basemaps/basemaps-component.jsx

```diff
@@ -67,21 +74,120 @@ class Basemaps extends React.PureComponent {
           onClick={e => e.stopPropagation()}
         >
           {item.label}
-          <Dropdown
-            className="landsat-selector"
-            theme="theme-dropdown-native-inline"
-            value={year}
-            options={landsatYears}
-            onChange={value =>
-              selectBasemap(basemaps.landsat, parseInt(value, 10))
-            }
-            native
-          />
+          <div className="basemaps-list-item-selectors">
+            <Dropdown
+              className="landsat-selector"
+              theme="theme-dropdown-native-inline"
+              value={year}
+              options={landsatYears}
+              onChange={value => selectBasemap(basemap, parseInt(value, 10))}
+              native
+            />
+          </div>
+        </span>
+      </button>
+    );
+  }
+
+  renderTooltipBasemap(item) {
+    const { selectBasemap, activeBasemap, basemaps } = this.props;
+    const basemap = basemaps[item.value]
+      ? basemaps[item.value]
+      : basemaps.planet;
+    const years = [2016, 2017, 2018].map(y => ({ value: y, label: y }));
+    const year = activeBasemap.year || 2018;
+    const month = parseInt(activeBasemap.month, 10) || 1;
+    const frequency = activeBasemap.frequency || 'monthly';
+    return (
+      <button
+        className="basemaps-list-item-button"
+        onClick={() => selectBasemap(basemap, year, month)}
+      >
+        <div
+          className="basemaps-list-item-image"
+          style={{
+            backgroundImage: `url(${item.image})`
+          }}
+        />
+        <span
+          className="basemaps-list-item-name"
+          onClick={e => e.stopPropagation()}
+        >
+          {item.label}
+          <div className="basemaps-list-item-selectors">
+            <Tooltip
+              className="basemaps-tooltip"
+              theme="light"
+              interactive
+              arrow
+              sticky
+              useContext
+              html={this.renderTooltipWindow({
+                month,
+                year,
+                years,
+                frequency,
+                onDateChange: (m, y) => selectBasemap(basemap, m, y)
+              })}
+            >
+              <span>{`${`0${month}`.slice(-2)}/${year}`}</span>
+            </Tooltip>
+          </div>
         </span>
       </button>
     );
   }
 
+  renderTooltipWindow(options) {
```

Comment:
This is rendering a very specific view with specific settings. Calling it tooltipWindow isnt really true. Make a reusable function for this. Inside the config for each basemap you can create a callback function for processing the data here.